domains/rpc/composables/useAssetRpc.ts

```diff
@@ -0,0 +1,211 @@
+
+export function useAssetRpc() {
+  return (
+    _address?: MaybeRef<Address | undefined>,
+    _tokenId?: MaybeRef<string | undefined>
+  ) => {
+    const connectedProfile = useProfile().connectedProfile()
+    const profileAddress = computed(() => connectedProfile.value?.address)
+    const { currentNetwork } = storeToRefs(useAppStore())
+    const queries: ComputedRef<
+      QFQueryOptions[] & {
+        address: Address | undefined
+        tokenId: string | undefined
+      }
+    > = computed(() => {
+      const chainId = currentNetwork.value?.chainId || ''
+      const tokenId = isRef(_tokenId) ? _tokenId.value : _tokenId
+      const address: Address | undefined = isRef(_address)
+        ? _address.value
+        : _address
+
+      const queries: QFQueryOptions[] & {
+        address: Address | undefined
+        tokenId: string | undefined
+      } = (
+        address
+          ? [
+              queryGetData({
+                // 0
+                chainId,
+                address,
+                keyName: 'LSP4TokenName',
+              }),
+              queryGetData({
+                // 1
+                chainId,
+                address,
+                keyName: 'LSP4TokenSymbol',
+              }),
+              queryGetData({
+                // 2
+                chainId,
+                address,
+                keyName: 'LSP4TokenType',
+              }),
+              queryGetData({
+                // 3
+                chainId,
+                address,
+                keyName: 'LSP8TokenMetadataBaseURI',
+              }),
+              queryGetData({
+                // 4
+                chainId,
+                address,
+                keyName: 'LSP8TokenIdFormat',
+              }),
+              queryCallContract({
+                // 5
+                chainId,
+                address,
+                method: 'totalSupply()',
+              }),
+              profileAddress.value
+                ? queryCallContract({
+                    // 6
+                    chainId,
+                    address,
+                    method: 'balanceOf(address)',
+                    args: [profileAddress.value],
+                    staleTime: 250,
+                  })
+                : queryNull(),
+              ...interfacesToCheck.map(({ interfaceId }) => {
+                return queryCallContract({
+                  // 7 / 8
+                  chainId,
+                  address: address || '0x0',
+                  method: 'supportsInterface(bytes4)',
+                  args: [interfaceId],
+                })
+              }),
+            ]
+          : []
+      ) as QFQueryOptions[] & {
+        address: Address | undefined
+        tokenId: string | undefined
+      }
+      queries.tokenId = tokenId
+      queries.address = address
+      return queries
+      // Trick to keep the right receivedAssetCount and allAddresses attached
+      // to the current queries list.
```

Comment:
Not sure what this comment is about. Looks like belongs to `domains/rpc/composables/useProfileAssetsRpc.ts` 🙂

Reply:
This is how we're indexing the output for the later combine.

Reply:
Yeah seem like some leftover, I'll ✂️

Reply:
But there are no `receivedAssetCount and allAddresses` in this file and it's the same comment as from `useProfileAssetsRpc.ts`. I think it is just in a wrong place.